### PR TITLE
Improve mobile checkout

### LINKED ISSUE #12626: Domains: No indication of "yearly" price in checkout on mobile
In a recent usability test on mobile, some users in checkout were confused about the total cost because there is no indication within checkout that the price is yearly. 

image

On >=660 screens this information is communicated in the "cart" sidebar: 
image

### PR DESCRIPTION
**Pull request by mikeshelton1503**
Fixes #12626 

This PR improves mobile checkout by:

- Providing an option to view the cart (now renamed "Order Summary") which was previously hidden at all times <660px
- Renaming "Cart" to "Order Summary"
- Updating to use `SectionHeader` component for Calypso consistency.
- General minor design improvements (alignment, type-size, padding, etc.)

Review:
- [x] Product
- [x] Code

### CODE REVIEW

File: client/my-sites/upgrades/checkout/cart-toggle.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+		const show = ! this.state.isShowingCartOnMobile;
+		this.setState( { isShowingCartOnMobile: show } );
+		showCartOnMobile( show );
+	};
+
+	render() {
+		const label = this.state.isShowingCartOnMobile
+			? this.props.translate( 'Hide order summary' )
+			: this.props.translate( 'Show order summary' );
```
Comment (a8ci18n) [inline client/my-sites/upgrades/checkout/cart-toggle.jsx]: Hi! I've found a possible matching string that has already been translated **29** times:
`translate( 'Order Summary' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/upgrades/checkout/cart-toggle.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+		event.preventDefault();
+
+		const show = ! this.state.isShowingCartOnMobile;
+		this.setState( { isShowingCartOnMobile: show } );
+		showCartOnMobile( show );
+	};
+
+	render() {
+		const label = this.state.isShowingCartOnMobile
+			? this.props.translate( 'Hide order summary' )
```
Comment (a8ci18n) [inline client/my-sites/upgrades/checkout/cart-toggle.jsx]: Hi! I've found a possible matching string that has already been translated **29** times:
`translate( 'Order Summary' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/upgrades/checkout/cart-toggle.jsx (hunk 1 above)
Comment (klimeryk) [inline client/my-sites/upgrades/checkout/cart-toggle.jsx]: Sorry little one - close, but no cigar.

**Comment by klimeryk** [comment]
@mikeshelton1503 I've implemented the toggling of the order summary on mobile - I'm adding/removing the `shopping-cart__hidden` style to secondary cart when the user clicks that link. The style only does anything for `<660px`, so non-mobile view should not be affected. Feel free to adjust or move/rename the style as needed.

I've noticed that the `SectionHeader` change you've introduced causes some issues with the animations we have for the cards. See how the headers are not animated when the form is being shown:
3a5kxvy00s

Not sure if this was intended, but it did stick out to me (maybe because I'm used to the previous look).

**Comment by mikeshelton1503** [comment]
Thanks @klimeryk!

I fixed the issue with the animation that you spotted. Good catch!

The only thing remaining is I noticed that when you click the show/hide toggle button the content gets pushed under the masterbar:

image

I couldn't pin point what was causing it. Any ideas? I can take another look tomorrow with fresh eyes.

**Comment by klimeryk** [comment]
I'm forcing the view to scroll to the cart when it's toggled to show - otherwise it seemed like it was easy to miss? You can disable this "feature" by removing this whole function: https://github.com/Automattic/wp-calypso/pull/13102/files#diff-fe4588cc8e39b88e740f8e477839947aR52 I've tested it with the credits payment box, but it might be causing problems for other payment types I guess?

**Comment by klimeryk** [comment]
@mikeshelton1503 Sorry, now I understand better what you were referring to. It seems it's not related to the scroll thing, but it's a more generic problem on mobile:
screen shot 2017-04-19 at 12 56 04
screen shot 2017-04-19 at 12 57 13

Maybe a missing margin that should be applied on mobile to offset the master bar on top?

**Comment by klimeryk** [comment]
@mikeshelton1503 I've fixed the bug with the scrolling.

But while testing it, I've noticed that on mobile the `x` buttons in the cart doesn't seem to work :man_facepalming: Could you confirm that you are seeing this too? Probably not intended ;/

**Comment by mikeshelton1503** [comment]
@klimeryk Good catch! I have fixed that bug now. I think we're good to go now.

**Comment by klimeryk** [comment]
I've made a small change to the PayPal payment box - to make its layout the same as the credit card and credits ones (so, the coupon and cart toggle after the primary button).

While doing some final testing, I noticed that the toggling logic I've introduced does not work too well when the cart is updated. It seems that when the cart is updated the components are removed and re-added to the DOM and that causes some conflicting state, resulting for example in showing "Show Order Summary" in the payment box, but the order summary is already visible below.
Working on fixing it now... ⌛️

**Comment by breezyskies** [comment]
Product 👍

File: client/my-sites/upgrades/checkout/style.scss [hunk 3]
```diff
@@ -809,9 +824,11 @@
 	display: block;
 	clear: both;
 	font-size: 12px;
+	padding-bottom: 5px;
 
 	@include breakpoint( "<660px" ) {
-		margin: 20px 0 0 0;
+	border-bottom: 1px solid lighten( $gray, 30% );
```
Comment (scruffian) [inline client/my-sites/upgrades/checkout/style.scss]: Missing tab here

**Comment by scruffian** [comment]
This looks good 👍

**Comment by klimeryk** [comment]
@breezyskies and @scruffian - I'm not sure I understand your feedback correctly - does it mean you can't reproduce the bug I've mentioned above: https://github.com/Automattic/wp-calypso/pull/13102#issuecomment-295786639 ? While I appreciate the feedback, I thought I made it clear that I'm working on fixing it before this PR could be reviewed and merged?

**Comment by klimeryk** [comment]
OK, _now_ it's ready for review ;)

**Pull request merged**